### PR TITLE
Macros

### PR DESCRIPTION
**Pull request by matthayes**
These were written by Will Vaughan.  Just checking them in.

I added several unit tests.  However these can't be run with pig 0.10 due to a bug in importing macros with PigUnit.  This will be fixed by pig 0.11.  For now they are disabled.  However they do pass with a patched pig 0.10.

### CODE REVIEW

File: test/pig/datafu/test/pig/macros/countByParallel.pig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+register $JAR_PATH
+
+IMPORT '$MACROS_PATH';
+
+data_in = LOAD 'input' as (key:chararray,value:int);
+
+data_out = count_by_p(data_in,key,5); -- parallel 5
```
Comment (samshah) [inline test/pig/datafu/test/pig/macros/countByParallel.pig]: No count_by_p defined, test won't work.

**Comment by samshah** [comment]
Why the need to include a binary pig 0.10 jar?

Also, how could we document macros?

File: test/pig/datafu/test/pig/macros/countByParallel.pig (hunk 1 above)
Comment (matthayes) [inline test/pig/datafu/test/pig/macros/countByParallel.pig]: File isn't used, forgot to delete it, now deleted in new commit

**Comment by matthayes** [comment]
I included the patched pig 0.10 jar for convenience to run the macros unit tests if necessary as we add more.

For documentation, perhaps we could have a convention where comments are placed above each macro, similar to java documentation, and we write a script which parses out the comments and created a markdown file?

**Comment by samshah** [comment]
As discussed, let's think it through and see how best to produce reusable macros with documentation.

**Pull request closed without merging**